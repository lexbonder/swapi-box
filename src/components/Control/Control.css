.Control {
  display: flex;
  justify-content: space-around;
  padding: 20px;
}

.Control button { 
  background-color: transparent;
  border: 1px solid #feda4a;
  color: #4A53FE;
  font-size: 40px;
  font-weight: bold;
  outline: none;
}

.Control button:hover {
  background-color: #feda4a;
}

button.clicked {
  background-color: #feda4a;
}

@media screen and (max-width: 970px) {
  .Control {
    width: 87%;
    margin: auto;
  }

  .Control button {
    margin: 10px;
    font-weight: normal;
  }
}

@media screen and (max-width: 775px) {
  .Control {
    width: 448px;
    flex-wrap: wrap;
  }
}

@media screen and (max-width: 468px) {
  .Control {
    width: 100%;
  }
}
