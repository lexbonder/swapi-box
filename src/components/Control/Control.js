import React from 'react';

const Control = ({ getCards }) => {
  return(
    <div>
        <button>View Favorites</button>
        <button onClick={getCards} name='people'>People</button>
        <button onClick={getCards} name='planets'>Planets</button>
        <button onClick={getCards} name='vehicles'>Vehicles</button>
    </div>
  )
}

export default Control;