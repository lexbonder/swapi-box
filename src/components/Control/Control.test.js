/* eslint-disable */
import React from 'react';
import {shallow} from 'enzyme';
import Control from './Control';

describe('Control', () => {
  let wrapper;
  let mockHandleClick = jest.fn()
  let mockEvent = {target: {name : 'people', classList: {add: () => {}}}}

  beforeEach(() => {
    wrapper = shallow(<Control handleClick={mockHandleClick} numFav={1}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call handle click when a button is pressed', () => {    
    wrapper.find('button').first().simulate('click', mockEvent)
    // wrapper.find('button').last().simulate('click')
    
    expect(mockHandleClick).toHaveBeenCalled()
  })
})