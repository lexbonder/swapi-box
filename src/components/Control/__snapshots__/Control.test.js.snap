// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Control
    handleClick={[Function]}
    numFav={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          name="favorites"
          onClick={[Function]}
>
          View Favorites (
          1
          )
</button>,
        <button
          name="people"
          onClick={[Function]}
>
          People
</button>,
        <button
          name="planets"
          onClick={[Function]}
>
          Planets
</button>,
        <button
          name="vehicles"
          onClick={[Function]}
>
          Vehicles
</button>,
      ],
      "className": "Control",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "View Favorites (",
            1,
            ")",
          ],
          "name": "favorites",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "View Favorites (",
          1,
          ")",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "name": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "name": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "name": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            name="favorites"
            onClick={[Function]}
>
            View Favorites (
            1
            )
</button>,
          <button
            name="people"
            onClick={[Function]}
>
            People
</button>,
          <button
            name="planets"
            onClick={[Function]}
>
            Planets
</button>,
          <button
            name="vehicles"
            onClick={[Function]}
>
            Vehicles
</button>,
        ],
        "className": "Control",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "View Favorites (",
              1,
              ")",
            ],
            "name": "favorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "View Favorites (",
            1,
            ")",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "name": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "name": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "name": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
